src/JWT/ValidationParameters.cs

```diff
@@ -1,38 +1,48 @@
 ﻿namespace JWT
 {
     /// <summary>
-    /// Contains a set of parameters that are used by a <see cref="JwtValidator"/> when validating a token.
+    /// Contains a set of parameters that are used by a <see cref="JwtValidator" /> when validating a token.
     /// </summary>
     public class ValidationParameters
     {
         /// <summary>
         /// Gets or sets a boolean that controls if validation of the signature that signed the token is called.
         /// </summary>
-        public bool ValidateIssuerSigningKey { get; set; } = true;
+        public bool ValidateIssuerSigningKey { get; set; }
 
         /// <summary>
         /// Gets or sets a boolean to control if the lifetime will be validated during token validation.
         /// </summary>
-        public bool ValidateLifetime { get; set; } = true;
+        public bool ValidateLifetime { get; set; }
 
         /// <summary>
         /// Gets or sets a boolean to control if the issued time will be validated during token validation.
         /// </summary>
-        public bool ValidateIssuedTime { get; set; } = true;
+        public bool ValidateIssuedTime { get; set; }

```

Comment:
Could these default values be added back? Creating a new `ValidationParameters` object with a single property value set to `false` (such as in the tests) without these default values would require a consumer to either define the value for each property:
```csharp
var valParams = new ValidationParameters
{
    ValidateIssuerSigningKey = true,
    ValidateLifetime = true,
    ValidateIssuedTime = false
};
```
or use the default property `ValidationParameters.Default` and then manually set the value of the property you want to change:
```csharp
var valParams = ValidationParameters.Default;
valParams.ValidateIssuedTime = false;
```
which seems a bit clunky.

Reply:
Do we want to introduce Fluent API for ValidationParameters to? It can be simply a set of extensions methods:
```c#
public ValidationParameters NotValidateIssuedTime(this ValidationParameters  valParams) =>
    valParams.ValidateIssuedTime(false);

public ValidationParameters ValidateIssuedTime(this ValidationParameters  valParams, bool value)
{
valParams.ValidateIssuedTime = value;
return valParams;
}
```

Reply:
I think that adding a fluent api for the `ValidationParameters` is not needed since it's just a class with a bunch of properties; using the fluent api would end up needing more code and would increase maintenance if any additional parameters were added.

However, I do think it should be clear what the default property values are.

Do you think the default value of the boolean properties should be `false` rather than `true`?

Reply:
Hey @binki, I'm sorry once again! I'm getting overloaded at work, coming back to your PR as soon as get a spare moment.

I was thinking more about the default values and setting them all to `true` is unusual and counterintuitive. Should we have flip all the properties so they all would have `false` by default so just `new` would be identical to `Default`? Same story for `margin` which is `0` by default already.

Also maybe we should add the ctor accepting `margin` back? And make it pass the value to the ctor of parameters

Reply:
@abatishchev I’m sure you meant to mention @nathanpovo (though GitHub thinks you shouldn’t mention him because he is the one who reported this issue). However, I still have opinions. Having initialization on properties is not strange for classes. Changing the parameter names to allow default values to make sense (e.g., having `DoNotValidateIssuerSigningKey` implicitly initialized to `false` instead of `ValidateIssuerSigningKey` initialized to `true`) is only needed for structs.

I think it would be most bad to make it easy to accidentally turn validation off entirely. So I’d prefer the properties to be initialized to what are considered good defaults.

If you want people to use `Default` instead, then you should make the constructor `private` so that no one can accidentally instantiate the class directly. This way, they would be forced to choose `None` or `Default` instead of being able to accidentally use `None` by calling the constructor.

Reply:
> manually set the value of the property you want to change:
> 
> ```cs
> var valParams = ValidationParameters.Default;
> valParams.ValidateIssuedTime = false;
> ```
> 
> which seems a bit clunky.

If you find this clunky, you can always define an extension method like the following if you ever want to make sure you have fully initialized an object before making it available to the current scope:

```
public static class ObjectExtensions {
  public static T With<T>(
    this T o,
    Action<T> action) {
    action(o);
    return o;
  }
}
```

Reply:
@abatishchev, no worries! The feature isn't urgently needed, I'm just adding it to the library to help out and to make it a bit easier to use.

@binki, thanks for the valuable input. I agree that it's important to not make it easy to turn off the validation by accident; that's why I had set the property values to `true` so that the validation would get done by default. Setting the constructor to `private` seems like the easiest thing to do; that way we could leave the defaults of the properties to whatever the default of the type is and also prevent a user from creating validation parameters that will not validate. We could then use an extension method, as suggested by @binki, to easily change the values from the default.

Reply:
Thanks, I am glad to see discussion is happening!

I think I must clarify/condition my suggestion to use an extension method just to make sure we are understanding each other ;-). I do not think it would be appropriate to add the exact extension method (`With<T>`) I suggested to this library. That is more of a personal coding style/choice. I define that in my own projects because sometimes I want to be able to stuff related things into an expression. For example, I might use it to initialize [`XmlWriterSettings`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlwritersettings?view=net-5.0). It belongs in a utility class or package for use by those who subscribe to this coding style.

What would be more appropriate would probably be a “fluent” API as discussed earlier. This could be as simple as calling a callback (like with my `With<T>`, but just targeted to the `ValidationParameters` type instead of generic).

Reply:
@binki sorry for involving you into the discussion outside your will :) and thanks for valuable input!
Totally agree that the most important aspect should be preventing accidentally turning off the validation, that would be disastrous.

Reply:
@nathanpovo generic `With<T>` is too generic in my opinion. it's better to add few `ValidationParameters`-specific methods. This way they'll be scoped, discoverable, and readable. what you think?
Sorry for going back and forth on this!

Reply:
Yeah, a generic method wouldn't make sense in this case; I added an extension method just to modify `ValidationParameters` objects.